src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,110 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private T[] arr;
```

Comment:
The use of an array `arr` to implement a linked list is not in line with the linked list data structure concept. Linked lists should consist of nodes that are linked together, not a fixed-size array.